### PR TITLE
EOS-14281: [LDR R1] Enable addb dump for s3server by default.

### PR DESCRIPTION
**Pull request by amitwac2608**
I have added a new config parameter in s3config.yaml to read it in s3server
startup script, and add "--addb" option, if the new conf parameter is set.

Signed-off-by: Amit Kumar <amit.kumar@seagate.com>

### CODE REVIEW

File: s3config-test.yaml [hunk 1]
```diff
@@ -69,6 +69,7 @@ S3_SERVER_CONFIG:                                     # Section for S3 Server
    S3_REDIS_SERVER_PORT: 6379                           # In case if redis is used for kvs contains redis server port
    S3_SERVER_MOTR_ETIMEDOUT_MAX_THRESHOLD: 100          # Number of ETIMEDOUT errors per monitoring window before s3server restart
    S3_SERVER_MOTR_ETIMEDOUT_WINDOW_SEC: 2               # Monitoring window for motr ETIMEDOUT errors in seconds
+   S3_SERVER_ENABLE_MOTR_ADDB_DUMP: true                # If set to true, then addb dump will be collected
```
Comment (dsurnin) [inline s3config-test.yaml:72]: it is not only motr addb - they are independent each other, so I think it is better to rename it

File: system/s3startsystem.sh [hunk 2]
```diff
@@ -72,22 +78,15 @@ profile_fid="<$MOTR_PROFILE_FID>"
...
+  addb_option="--addb"
+fi
 
 pid_filename='/var/run/s3server.'$1'.pid'
 set -x
 s3server --s3pidfile $pid_filename \
          --motrlocal $local_ep --motrha $ha_ep \
          --motrprofilefid $profile_fid --motrprocessfid $process_fid \
-         --s3port $s3port --log_dir $s3_log_dir
+         --s3port $s3port --log_dir $s3_log_dir $addb_option
```
Comment (dsurnin) [inline system/s3startsystem.sh:92]: I would prefer to have this options being processed directly in s3server like any other config params

**Review by dsurnin — APPROVED** [review]

File: s3config-test.yaml (hunk 1 above)
Comment (amitwac2608) [reply s3config-test.yaml]: done.

File: system/s3startsystem.sh (hunk 2 above)
Comment (amitwac2608) [reply system/s3startsystem.sh]: done.

**Comment by amitwac2608** [comment]
Jenkins Job link: http://eos-jenkins.colo.seagate.com/job/Cortx-PR-Build/job/S3server-7.8.2003/17

This job does not include UT commit. I will run new job again, once current job finishes.

**Comment by amitwac2608** [comment]
Jenkins Job link: http://eos-jenkins.colo.seagate.com/job/Cortx-PR-Build/job/S3server-7.8.2003/17

This job does not include UT commit. I will run new job again, once current job finishes.

> 
> 
> Jenkins Job link: http://eos-jenkins.colo.seagate.com/job/Cortx-PR-Build/job/S3server-7.8.2003/17
> 
> This job does not include UT commit. I will run new job again, once current job finishes.

Job#17 passed.
Fired new job which has latest commitID: http://eos-jenkins.colo.seagate.com/job/Cortx-PR-Build/job/S3server-7.8.2003/23

File: dev-starts3.sh [hunk 3]
```diff
@@ -197,6 +197,6 @@ do
   pid_filename='/var/run/s3server.'$s3port'.pid'
   $callgraph_cmd s3server --s3pidfile $pid_filename \
            --motrlocal $local_ep:${motr_local_port} --motrha $ha_ep \
-           --s3port $s3port --fault_injection true $fake_params --loading_indicators --addb --getoid true
+           --s3port $s3port --fault_injection true $fake_params --loading_indicators --getoid true
```
Comment (knrajnambiar76) [inline dev-starts3.sh:200]: for dev VM env its disabled right

File: s3config.yaml [hunk 4]
```diff
@@ -69,6 +69,7 @@ S3_SERVER_CONFIG:
    S3_REDIS_SERVER_PORT: 6379                           # In case if redis is used for kvs contains redis server port
    S3_SERVER_MOTR_ETIMEDOUT_MAX_THRESHOLD: 100          # Number of ETIMEDOUT errors per monitoring window before s3server restart
    S3_SERVER_MOTR_ETIMEDOUT_WINDOW_SEC: 1               # Monitoring window for motr ETIMEDOUT errors in seconds
+   S3_SERVER_ENABLE_ADDB_DUMP: true                     # If set to true, then addb dump will be collected
```
Comment (knrajnambiar76) [inline s3config.yaml:72]: On Dev environment, do we need it to be enabled by default ?

**Review by knrajnambiar76 — APPROVED** [review]

File: s3config.yaml (hunk 4 above)
Comment (amitwac2608) [reply s3config.yaml:72]: It was enabled in dev environment (in startup script, we were passing --addb option) previously as well. SO kept it same.

Should we set it to False for dev env ?

Comment (knrajnambiar76) [reply s3config.yaml:72]: Ok, let it be like that. 
Second thought says that we may be able to catch some issue in Dev itself.

**Comment by amitwac2608** [comment]
> 
> 
> Jenkins Job link: http://eos-jenkins.colo.seagate.com/job/Cortx-PR-Build/job/S3server-7.8.2003/17
> 
> This job does not include UT commit. I will run new job again, once current job finishes.
> 
> > Jenkins Job link: http://eos-jenkins.colo.seagate.com/job/Cortx-PR-Build/job/S3server-7.8.2003/17
> > This job does not include UT commit. I will run new job again, once current job finishes.
> 
> **Job#17 passed.**
> Fired new job which has latest commitID: http://eos-jenkins.colo.seagate.com/job/Cortx-PR-Build/job/S3server-7.8.2003/23

**The latest job also passed.** PR is fit to merge.

**Pull request merged**